### PR TITLE
fix: tag not using elipses when width constrained

### LINKED ISSUE #4511: cv-tag filter breaks when using text with length bigger than the width of the tag
## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-vue`

## Detailed description

> Describe in detail the issue you're having.

Our team uses tags to present content generated by others. The problem we have is long text in tags doesn’t seem to be handled correctly. 

> Is this issue related to a specific component?

yes

> What did you expect to happen? What happened instead? What would you like to
> see changed?

Expected: To add ellipsis syntax (3 dots like `...`) and `x` icon to remain same size
Happens: Keeps the text in the screen and `x` icon gets smaller

> What browser are you working in?

Firefox 

> What version of the Carbon Design System are you using?

Latest

## Steps to reproduce the issue

1. Open vue storybook for cv-tag filter (http://vue.carbondesignsystem.com/?path=/story/components-cvtag--filter). Also tested in react storybook
2. Set the `tag label` to `South Georgia and the South Sandwich Islands`
3. Resize the width of the window (using the `Responsive Design Mode` from the dev tools or otherwise) to emulate a smaller div where this tag would be placed in

## Additional information

- Screenshot

A screenshot can be seen below from the vue storybook (I’ve highlighted the text using my mouse):

image

### PR DESCRIPTION
**Pull request by lee-chase**
Closes #4511

Makes the tag/filter to show the text elipses when the tag is too large for the 
#### Changelog

**Changed**

M       packages/components/src/components/tag/_tag.scss
M       packages/components/src/components/tag/tag.hbs

#### Testing / Reviewing

Ran up base components storybook for both filter and tag. Used developer tools to constrain the width of the containing div.

### CODE REVIEW

File: packages/components/src/components/tag/_tag.scss [hunk 1]
```diff
@@ -81,14 +84,26 @@
...
+    text-overflow: ellipsis;
+  }
+
   // tags used for filtering
   .#{$prefix}--tag--filter {
     @include tag-theme($inverse-02, $inverse-01);
+
     cursor: pointer;
-    padding-right: rem(2px); // Align with hover circle of X button
+    padding-right: rem(26px); // Space for hover circle and X button
```
Comment (emyarod) [inline packages/components/src/components/tag/_tag.scss]: just to confirm the 26px is from 20px width svg, 2px left border, 2px right border, and 2px space from the right edge right? should we express it as this?
```suggestion
    padding-right: calc(#{$carbon--spacing-06} + #{rem(2px)}); // icon width + 2px space from right edge
```

**Review by emyarod — APPROVED** [review]
looks good to me pending design approval

filter

image

standard

image

**Review by jendowns — COMMENTED** [review]
a11y question --

If/when text is truncated, how would someone be able to see/access the full text of the tag?

File: packages/components/src/components/tag/_tag.scss (hunk 1 above)
Comment (joshblack) [reply packages/components/src/components/tag/_tag.scss]: bump @lee-chase if you have a sec!

**Comment by joshblack** [comment]
@jendowns if I understand 2.1 requirements correctly, truncated text will need a tooltip. For 2.0, I believe it needs the `title` prop.

One question I had for this, do we want to go down the route of truncated text in the tag or do we just want it to scale as the text overflows? Seems like making sure it scales would help in our text resize situations but let me know if I'm missing something here!

**Comment by laurenmrice** [comment]
To joshs comment my gut feeling is I don't think we should have truncated text inside a tag. I think what is in the tag should be visible at all times.  But maybe there are some use cases for it so we could have it.

**Comment by aagonzales** [comment]
IMO Tags by default should not have a max width, they should just keep growing until the edge of the container. 

However, tags should also never be multi-lined so if a tag is constrained by space then it should elipse with a tooltip to show the full content. 

image

**Comment by aagonzales** [comment]
I can really only see this being a problem with user generated tags. Any system tags shouldn't have to ellipse, designers should be putting tags in layout/content area's that allow for whatever the tag widths are needed or designers should consider alternatives for displaying this information if size constraints is problematic.

**Comment by skaparelos1** [comment]
Hello,
in case you need data to help your decision: our team's case (IBM team) falls in the user-generated tags bucket. We are using data taken from an API we don't own that could be of variable size. Also the data we get from the API is subject to changing at some point.

**Comment by aagonzales** [comment]
@skaparelos1 why is adding an elipses better than letting the tag be displayed at full width? At what max size were you thinking this tags will elipse? 

Can you also share a screenshot or where these tags are being used so we can understand the content? Thanks!

**Comment by skaparelos1** [comment]
> why is adding an elipses better than letting the tag be displayed at full width?

If the user selected a location like `South Georgia and the South Sandwich Islands` this would need a lot of space even on a big screen. What if the user is using a mobile phone? 

> At what max size were you thinking this tags will elipse? 

Ideally I would like this to depend on the size of the parent. If it doesn't fit in the parent, then add ellipsis. If it does, then show it all.

> Can you also share a screenshot or where these tags are being used so we can understand the content? 

I am not sure if I am allowed to post a screenshot here. These tags are being used as filter refinements for a search application. Center of page contains search results whereas on the left side users are given the option to select several options to further refine their search.

**Comment by aagonzales** [comment]
Ok let's add the elipse function but the tag itself shouldn't have a max width on it (if that makes since). I think we'll need to be careful on the design side when this can be used.

**Review by designertyler — APPROVED** [review]

**Comment by joshblack** [comment]
bump @dakahn what are the 2.1 requirements for ellipses/truncated text? Anything we need to consider here for ellipses in tags?

**Comment by dakahn** [comment]
I can't find a reference to any requirements around truncated text. Usability is a concern, but as long as we've addressed those it seems like [Checkpoint 1.4.4](https://www.ibm.com/able/guidelines/ci162/resize_text.html) relates to text truncation/rewrap _on resize_. 👍

**Review by dakahn — APPROVED** [review]

**Pull request merged**